Src/Newtonsoft.Json/Utilities/MiscellaneousUtils.cs

```diff
@@ -40,6 +40,37 @@ namespace Newtonsoft.Json.Utilities
 
     internal static class MiscellaneousUtils
     {
+        internal const string TrimWarning = "Newtonsoft.Json relies on reflection over types that may be removed when trimming.";
+
+        internal const string AotWarning = "Newtonsoft.Json relies on dynamically creating types that may not be available with Ahead of Time compilation.";
+
+#if HAVE_APPCONTEXT
+        [FeatureSwitchDefinition("Newtonsoft.Json.SerializationIsSupported")]
+        [FeatureGuard(typeof(RequiresUnreferencedCodeAttribute))]
+        [FeatureGuard(typeof(RequiresDynamicCodeAttribute))]
+        internal static bool SerializationIsSupported => AppContext.TryGetSwitch("Newtonsoft.Json.SerializationIsSupported", out bool isSupported) ? isSupported : true;
```

Comment:
My suggestion on these names is to name them like public .NET APIs. That way if we ever needed to expose this as a public property, we could match the signature to the name.

For example https://github.com/dotnet/runtime/blob/128fdfd1fd982e277e969d5c7ca1710345f4f649/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Helpers.cs#L27